controller/aks-cluster-config-handler.go

```diff
@@ -707,21 +709,36 @@ func (h *Handler) updateUpstreamClusterState(ctx context.Context, secretsCache w
 		return config, err
 	}
 
-	// check tags for update
-	if config.Spec.Tags != nil {
+	// check tags for update. Skip the tags update if there was recently an unexpected response from Azure for a tags
+	// update so the cluster can become active.
+	if config.Spec.Tags != nil && !UnexpectedTagUpdateResponseInAzure {
+
 		if !reflect.DeepEqual(config.Spec.Tags, upstreamSpec.Tags) {
 			logrus.Infof("Updating tags for cluster [%s]", config.Spec.ClusterName)
 			tags := containerservice.TagsObject{
 				Tags: *to.StringMapPtr(config.Spec.Tags),
 			}
-			_, err = resourceClusterClient.UpdateTags(ctx, config.Spec.ResourceGroup, config.Spec.ClusterName, tags)
+			response, err := resourceClusterClient.UpdateTags(ctx, config.Spec.ResourceGroup, config.Spec.ClusterName, tags)
 			if err != nil {
 				return config, err
 			}
+			// Azure will regenerate default upstream tags and add them to the cluster resource if a user tries to
+			// remove them all in one request, but the Azure job that removes tags doesn't return an error in that
+			// case. So, we check whether Azure returns an unexpected state for the tags and if so, log the response
+			// and move on. Any upstream tags that were regenerated and put back on the cluster by Azure will be synced
+			// back to rancher.
+			if !reflect.DeepEqual(tags, upstreamSpec.Tags) && response.Status() == "Succeeded" {
```

Comment:
Updates to objects in Azure are done asynchronously. It seems that the tags call is very quick, but is it possible to get to this line and have `response.Status() == ""`?

Reply:
No I don't think so. Rancher waits for the tags call to return with a valid API response before moving on from the `UpdateTags` line.

Reply:
The response is a `Future` which means it doesn't wait for the action to complete. The API response is returned, but the `.Status()` call is on the `Future`.

Reply:
I didn't know that, thank you. Unsure if `response.Status()` could return "", but I updated the func to check the latest HTTP response code of the async call. That will most likely return a valid response code instead of possibly returning an empty string.

This is straight from the Azure SDK

```
// returns the cached HTTP response after a call to pollForStatus(), can be nil
latestResponse() *http.Response
```